packages/flutter/lib/src/widgets/selectable_region.dart

```diff
@@ -463,6 +467,8 @@ class SelectableRegionState extends State<SelectableRegion> with TextSelectionDe
         // case we want to retain the selection so it remains when we return to
         // the Flutter application.
         clearSelection();
+        _selectionStatusNotifier.value = SelectableRegionSelectionStatus.changing;
```

Comment:
also will this be easier to set changing in a certral place just before dispatchSelectionEvent?

Reply:
**Spent some time thinking about this, if we move the `changing` status update to before the `dispatchSelectionEvent`:**

If users are listening to the `SelectableRegionStatus`, and they check the selection in `SelectionListenerNotifier`, then when “changing” is called by `SelectableRegion` the selection will not be up to date yet. Only on “finalized” will it be up to date.

If users are listening to the `SelectionListenerNotifier`, then for example on a collapse, the status on the clear event notification will be “changing”, and on the end event notification it will also still be “changing”. This is because “finalized” is set by `SelectableRegion` after the end event notification.

**How it works now with `changing` status update after `dispatchSelectionEvent`:**

If users are listening to the `SelectableRegionStatus`, and they check the selection in `SelectionListenerNotifier`, then when “changing” is called the selection in the `SelectionListenerNotifier` is up to date, and on “finalized” it is also up to date.

If users are listening to the `SelectionListenerNotifier`, then for example on a collapse, the status on the clear event will be “finalized”, and on the end event notification it will also be “finalized” because it has not been updated from its default. `SelectableRegion` sets the “changed”/“finalized” status after the end event geometry change triggers a notification for listeners, so when listening to only `SelectionListenerNotifier` we would not get the finalized status.

**What should be the correct behavior:** I think the current behavior is correct, for listening to `SelectableRegionStatus` where the selection in `SelectionListenerNotifier` is up to date whenever `SelectableRegion` notifies. Though incorrect, when only listening to `SelectionListenerNotifier`, because the `SelectableRegionStatus` is not up to date when `SelectionListenerNotifier` notifies its listeners, for single taps its always going to be at the default "finalized".

`SelectionListenerNotifier` will always be behind, since it doesn’t reach the “finalized” at all, but it should have “changing” as the status when the `SelectionListenerNotifier` notifies its listeners of a selection change.

**Maybe we should have changing, changed, and finalized?**

changing, indicates the `SelectableRegion` will begin updating the selection. It will go before the `dispatchSelectionEvent` call.
changed, indicates the `SelectableRegion` has changed the selection. This will go after the `dispatchSelectionEvent` call.
finalized, indicates the `SelectableRegion` has finished updating the selection, this will go after the `dispatchSelectionEvent` call and after the “changed” update.

So when listening to `SelectableRegionStatus`, and a user checks the selection in `SelectionListenerNotifier`, then when “changing” is called the selection will be up to date (have the initial selection before any changes), and when “changed” is called the selection will also be up to date, and so will “finalized”.

When listening to only `SelectionListenerNotifier`, and a user checks the status in `SelectableRegion`, then for example on a collapse, the status on the clear event will be “changing”, and on the end event notification it will also be “changing”. `SelectableRegion` sets the “changed”/“finalized” status after the end event geometry change triggers a notification for listeners, so when listening to only `SelectionListenerNotifier` we would not reach those notifications.

Do you have any thoughts on this behavior?

Reply:
Having the updates in a central place is tricky because we don't want `SelectableRegion` to notify listeners on every `dispatchSelectionEvent` for example, for collapsing the selection we send two events, we only want to notify after the last event.

Reply:
Thanks for explanation, I think you are right. it is probably better to put it after the dispatch unless we want to do wiiChange, didChange style notification, which may not be necessary.

This LGTM

Reply:
Sounds good to me, talked offline and we will retain the current behavior, and expand on it if needed.